### ISSUE TITLE
0.4.0 breaks app, results in "Unhandled promise rejection TypeError"

### ISSUE THREAD

**Issue by davidluhr** [ChristopherBiscardi/gatsby-mdx#293]
**Describe the bug**
When using `"gatsby-mdx": "^0.4.0",`, Gatsby compiles a site successfully, but nothing renders in the DOM.

The console shows the following error:
```
Unhandled promise rejection TypeError: "_loaders_mdx_components__WEBPACK_IMPORTED_MODULE_9__.plugins is undefined"
  WrapRootElement wrap-root-element.js:41
  results api-runner-browser.js:29
  apiRunner api-runner-browser.js:20
  js root.js:110
  js commons.js:3122
  Bootstrap 6 Webpack
```

**To Reproduce**
1. Using https://github.com/ChristopherBiscardi/gatsby-mdx-minimal-repro, update `gatsby-mdx` to the latest version (`^0.4.0` at time of writing) in `package.json`
2. Run `npm i`, followed by `gatsby develop`
3. Observe that nothing renders to the DOM and the above console error appears.

**Expected behavior**
The app should build and render successfully without causing this error.

**Additional context**
I've tried all other combinations of dependencies in this repo to isolate the cause to `gatsby-mdx`. All other dependencies can be updated to their latest version and the app builds as expected, but if `gatsby-mdx` uses version `^0.4.0`, the app breaks.

This issue is occurring on my Windows 10 machine. I'm not sure if the same issue occurs on macOS. I've replicated this issue multiple Gatsby projects that use `gatsby-mdx`.

**Comment by johno** [comment]
Thanks for opening up this issue! I'm unable to reproduce on MacOS. It's looking like it might be related to Windows.

We'll keep digging.

**Comment by ChristopherBiscardi** [comment]
It has to be something about the new plugins functionality on windows: https://github.com/ChristopherBiscardi/gatsby-mdx/blob/70ee18f40fc02247a90b9f2619f62577b6a46656/packages/gatsby-mdx/loaders/mdx-components.js likely the way the webpack loader is defined interacting with windows pathing: https://github.com/ChristopherBiscardi/gatsby-mdx/blob/70ee18f40fc02247a90b9f2619f62577b6a46656/packages/gatsby-mdx/gatsby/create-webpack-config.js#L34

**Comment by davidluhr** [comment]
Thank you both for digging into this! I figured it was specific to Windows, but wasn't sure how to start investigating.

**Comment by ChristopherBiscardi** [comment]
fixed in 0.4.2

**Comment by davidluhr** [comment]
@ChristopherBiscardi Thank you so much! Very excited to get the latest running on my Windows machine!
